flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartition.java

```diff
@@ -118,13 +117,19 @@ protected Throwable getFailureCause() {
 
 	/**
 	 * Gets the number of non-event buffers in this subpartition.
-	 *
-	 * <p><strong>Beware:</strong> This method should only be used in tests in non-concurrent access
-	 * scenarios since it does not make any concurrency guarantees.
 	 */
-	@VisibleForTesting
-	public int getBuffersInBacklog() {
-		return buffersInBacklog;
+	public abstract int getBuffersInBacklog();
+
+	/**
+	 * @param lastBufferAvailable whether the last buffer in this subpartition is available for consumption
+	 * @return the number of non-event buffers in this subpartition
+	 */
+	protected int getBuffersInBacklog(boolean lastBufferAvailable) {
```

Comment:
Hmm, I think I would keep the `Unsafe` suffix here, since this method is not synchronised compared to the `decreaseBuffersInBacklog`. Ditto for the abstract?

Reply:
Yes, it makes sense to keep the `Unsafe` here consistent with `decreaseBuffersInBacklog`.

In another aspect, the existence of `Unsafe` is for distinguishing the different usages compared with `safe`. For `getBuffersInBacklog` there might need only one way for use either `safe` or `Unsafe`. Anyway I am willing to make it `Unsafe` here, then we could avoid explaining it in method comments.

Reply:
I get your point, however the two usages of this method below showed that it's confusing as it is now. 
> Anyway I am willing to make it Unsafe here, then we could avoid explaining it in method comments.

Making code self explanatory is definitely a good thing :)